Lib/posixpath.py

```diff
@@ -197,31 +194,19 @@ def ismount(path):
     except (OSError, ValueError):
         # It doesn't exist -- so not a mount point. :-)
         return False
-    else:
-        # A symlink can never be a mount point
-        if stat.S_ISLNK(s1.st_mode):
-            return False
 
-    path = os.fspath(path)
-    if isinstance(path, bytes):
-        parent = join(path, b'..')
-    else:
-        parent = join(path, '..')
-    parent = realpath(parent)
+    # A symlink can never be a mount point
+    if stat.S_ISLNK(s1.st_mode):
+        return False
+
+    parent = realpath(dirname(path))
```

Comment:
I don't think this works when `path` is empty.

Reply:
Now compare that with the result of the existing code, which I expect will give you `'C:\Users\wanne'`, i.e. the **parent** of the current working directory.

Reply:
```python
>>> import os
>>> os.lstat("")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
FileNotFoundError: [Errno 2] No such file or directory: ''
```

But it doesn't work for `.`:

```python
>>> ismount1("."), ismount2(".")
(False, True)
```

I believe this should work though (doesn't have problems with directories without execute permission):

```python
parent = dirname(abspath(path))
```